### PR TITLE
Add event clearing to `prefer-add-event-listener` rule

### LINKED ISSUE #163: The `prefer-add-event-listener` rule should not autofix event clearing
These should be reported, but not auto-fixed:

```js
foo.onclick = undefined;
```

```js
foo.onclick = null;
```

These need to be handled manually, as `removeEventListener` requires the listener function be able to unsubscribe.

### PR DESCRIPTION
**Pull request by medusalix**
Adds reporting functionality for these cases:

```js
foo.onclick = undefined;
```
```js
foo.onclick = null;
```

Resolves #163

### CODE REVIEW

**Comment by sindresorhus** [comment]
Yeah, I agree. Just add a test to ensure it's ignored then.

@futpib Unless you know a good way or module that can help with this?

**Comment by sindresorhus** [comment]
Can you fix the merge conflict?

File: test/prefer-add-event-listener.js [hunk 1]
```diff
@@ -26,10 +26,10 @@ const expectedBeforeUnloadWithReturnMessage = [
 ruleTester.run('prefer-add-event-listener', rule, {
 	valid: [
 		'foo.addEventListener(\'click\', () => {})',
+		'foo.removeEventListener(\'click\', onClick)',
 		'foo.onclick',
 		'foo.setCallBack = () => {console.log(\'foo\')}',
 		'setCallBack = () => {console.log(\'foo\')}',
-		'foo.onclick.bar = () => {}',
```
Comment (futpib) [inline test/prefer-add-event-listener.js]: Why is this no longer valid?

Comment (medusalix) [reply test/prefer-add-event-listener.js]: Oops, I think I might have deleted it by accident 😌.

**Comment by medusalix** [comment]
Any news on the review process?

**Comment by futpib** [comment]
I wonder about interaction between this and #218, could you also add a test for `onbeforeunload` clearing?

**Comment by medusalix** [comment]
It seems like the tests are working 😃. I had to change the code a little bit, but it looks like the other test cases are still behaving as before 👍.

**Comment by sindresorhus** [comment]
This looks good. Thank you, @medusalix! :)

**Pull request merged**